File: data-structures-and-algorithms-ashrafi/python/src/recursive/problem_8/problem_8.py
def print_truth_table(n, s):
    if len(s) == n:
        print(s)
    else:
        print_truth_table(n, s + "0")
        print_truth_table(n, s + "1")


def main():
    n = 2
    s = ""
    
    print_truth_table(n, s)


if __name__ == "__main__":
    main()

File: data-structures-and-algorithms-ashrafi/java/src/recursive/problem8/Problem8.java
public class Problem8 {
    
    public static void printTruthTable(int n, String s){
        if (s.length() == n) {
            System.out.print(s + "\n");
        } else{
            printTruthTable(n, s + "0");
            printTruthTable(n, s + "1");
        }
    }

    public static void main(String[] args) {
        int n = 2;
        String s = "";

        printTruthTable(n, s);
    }
}
